apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/pagerules/PageRuleServiceImpl.java

```diff
@@ -817,4 +821,42 @@ public ViewRuleResponse getRuleResponse(Long ruleId) {
                 ruleMetadata.getRuleDescText(), ruleMetadata.getSourceQuestionIdentifier(), sourceValues,
                 ruleMetadata.getLogic(), ruleMetadata.getTargetType(), ruleMetadata.getErrormsgText(), targetValues);
     }
+
+    @Override
+    public Page<ViewRuleResponse> getAllPageRule(Pageable pageRequest) {
+        Page<WaRuleMetadata> ruleMetadataPage =waRuleMetaDataRepository.findAll(pageRequest);
+
+        List<ViewRuleResponse> ruleMetadata =
+                ruleMetadataPage.getContent().stream().map((rule) ->new ViewRuleResponse(rule.getId(), rule.getWaTemplateUid(), rule.getRuleCd(),
+                rule.getRuleDescText(), rule.getSourceQuestionIdentifier(), buildSourceTargetValues(rule,true),
+                rule.getLogic(), rule.getTargetType(), rule.getErrormsgText(),
+                buildSourceTargetValues(rule,false))).collect(Collectors.toList());
+        return new PageImpl<>(ruleMetadata,ruleMetadataPage.getPageable(),ruleMetadata.size());
+
+    }
+
+    private List<String> buildSourceTargetValues(WaRuleMetadata ruleMetadata, Boolean isSource) {
+
+        List<String> sourceValues = new ArrayList<>();
+        List<String> targetValues = new ArrayList<>();
+        if(isSource){
+            if (ruleMetadata.getSourceValues() == null || ruleMetadata.getTargetQuestionIdentifier() == null) {
+                sourceValues.add(null);
+                return  sourceValues;
+            } else {
+                String[] sourceValue = ruleMetadata.getSourceValues().split(",");
+                return  Arrays.asList(sourceValue);
+            }
+        }else{
+            if (ruleMetadata.getSourceValues() == null || ruleMetadata.getTargetQuestionIdentifier() == null) {
+                targetValues.add(null);
```

Comment:
I think i prefer having an empty array instead of [null]. Is there a reason to return a null entry?